src/commands/close.rs

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Radical HQ Limited
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+use indoc::formatdoc;
+
+use crate::{
+    error::{Error, Result},
+    github::{PullRequestState, PullRequestUpdate},
+    output::{output, write_commit_title},
+};
+
+pub async fn close(
+    git: &crate::git::Git,
+    gh: &mut crate::github::GitHub,
+    config: &crate::config::Config,
+) -> Result<()> {
+    let mut prepared_commits = git.get_prepared_commits(config)?;
+
+    let prepared_commit = match prepared_commits.last_mut() {
+        Some(c) => c,
+        None => {
+            output("👋", "Branch is empty - nothing to do. Good bye!")?;
+            return Ok(());
+        }
+    };
+
+    write_commit_title(prepared_commit)?;
+
+    let pull_request_number =
+        if let Some(number) = prepared_commit.pull_request_number {
+            output("#️⃣ ", &format!("Pull Request #{}", number))?;
+            number
+        } else {
+            return Err(Error::new(
+                "This commit does not refer to a Pull Request.",
+            ));
+        };
+
+    // Load Pull Request information
+    let pull_request = gh.get_pull_request(pull_request_number).await??;
+
+    if pull_request.state != PullRequestState::Open {
+        return Err(Error::new(formatdoc!(
+            "This Pull Request is already closed!",
+        )));
+    }
+
+    output("📖", "Getting started...")?;
+
+    let base_is_master = pull_request.base.is_master_branch();
+
+    let result = gh
+        .update_pull_request(
+            pull_request_number,
+            PullRequestUpdate {
+                state: Some(PullRequestState::Closed),
+                ..Default::default()
+            },
+        )
+        .await;
+
+    match result {
+        Ok(closed) => closed,
+        Err(error) => {
+            output("❌", "GitHub Pull Request close failed")?;
+
+            return Err(error);
+        }
+    };
+
+    output("📕", "Closed!")?;
+
+    let mut remove_old_branch_child_process =
+        async_process::Command::new("git")
```

Comment:
I just landed #87 - you will have to rebase your PR and change `async::process` to `tokio::process`. See https://github.com/getcord/spr/pull/87/files#diff-e0cf2851ba3ab9d47423403ab3d5b29812959994245d5ca612a06b92629d608fL361-L388 for the details.

Reply:
Thanks for the heads up!